pkg/security/resolvers/process/resolver_windows.go

```diff
@@ -0,0 +1,77 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+//go:build windows
+// +build windows
+
+package process
+
+import (
+	"strings"
+	"sync"
+
+	"github.com/DataDog/datadog-agent/pkg/security/config"
+	"github.com/DataDog/datadog-agent/pkg/security/secl/model"
+	"github.com/DataDog/datadog-go/v5/statsd"
+)
+
+type Pid uint32
+type ProcessResolver struct {
+	maplock   sync.Mutex
+	processes map[Pid]*model.ProcessCacheEntry
+	opts      ProcessResolverOpts
+}
+
+// ProcessResolverOpts options of resolver
+type ProcessResolverOpts struct {
+}
+
+// NewProcessResolver returns a new process resolver
+func NewResolver(config *config.Config, statsdClient statsd.ClientInterface,
+	opts ProcessResolverOpts) (*ProcessResolver, error) {
+
+	p := &ProcessResolver{
+		processes: make(map[Pid]*model.ProcessCacheEntry),
+		opts:      opts,
+	}
+
+	return p, nil
+}
+
+// NewProcessResolverOpts returns a new set of process resolver options
+func NewResolverOpts() ProcessResolverOpts {
+	return ProcessResolverOpts{}
+}
+
+func (p *ProcessResolver) AddNewProcessEntry(pid Pid, file string, commandLine string) (*model.ProcessCacheEntry, error) {
+	e := model.NewEmptyProcessCacheEntry(uint32(pid), 0, false)
+
+	e.Process.PIDContext.Pid = uint32(e.Pid)
+	e.Process.Argv0 = file
+	e.Process.Argv = strings.Split(commandLine, " ")
```

Comment:
I'm concerned about splitting the command line on space here without at least taking quoting rules into account.

[process-agent](https://github.com/DataDog/datadog-agent/blob/4bd973b1fec5ef621a86d9125c775880dda200ea/pkg/process/procutil/process_windows.go#L575-L618) has some custom parsing rules

Windows provides [CommandLineToArgv](https://learn.microsoft.com/en-us/windows/win32/api/shellapi/nf-shellapi-commandlinetoargvw). But there's some discussion about the use of this method [here](https://github.com/golang/go/issues/15588).

Since the Windows CreateProcess APIs take a commandline and leave it up to the application to parse it as it will, some apps (like cmd.exe) have custom parsing rules, so we may not be able to have a one sized fits all split/unsplit for args.

Reply:
Point is well taken.
However, point of this PR is to get something in so we're not always chasing merge conflicts, etc.
Note that the PR doesn't include the driver, so there's nothing feeding this at this point.  This is more foundational.

I can certainly add a large comment here which indicates exactly how naive this implementation is, so it could be enhanced as we add functionality.

Reply:
At a minimum I think we need the comment; however, if there are straightforward / preferred alternatives (like the one Branden linked from process-agent) that improve the foundation from the start, I think we should consider using them to mitigate the risk of "we'll come back to this later" but not actually doing so.

Is there a reason we can't / shouldn't use the code from process-agent that Branden linked? That seems like it'd provide consistency throughout the agent for how the Windows command lines will be parsed and eventually represented in the UI. It'd also make it easy for process-agent to compare / contrast this implementation with their own if/when they shift to this implementation.